wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -51,16 +57,65 @@ import Network.Wai.Logger
 import Network.Wai.Middleware.RequestLogger.Internal
 import Network.Wai.Header (contentLength)
 import Data.Text.Encoding (decodeUtf8')
+import Network.Wai (Request)
 
 -- | The logging format.
 data OutputFormat
   = Apache IPAddrSource
+  | ApacheWithSettings ApacheSettings -- ^ @since 3.1.8
   | Detailed Bool -- ^ use colors?
   | DetailedWithSettings DetailedSettings -- ^ @since 3.1.3
   | CustomOutputFormat OutputFormatter
   | CustomOutputFormatWithDetails OutputFormatterWithDetails
   | CustomOutputFormatWithDetailsAndHeaders OutputFormatterWithDetailsAndHeaders
 
+-- | Settings for the `ApacheWithSettings` `OutputFormat`. This is purposely kept as an abstract data
+-- type so that new settings can be added without breaking backwards
+-- compatibility. In order to create an 'ApacheSettings' value, use 'defaultApacheSettings'
+-- and the various \'setApache\' functions to modify individual fields. For example:
+--
+-- > setApacheIPAddrSource FromHeader defaultApacheSettings
+--
+-- @since 3.1.8
+data ApacheSettings = ApacheSettings
+    { apacheIPAddrSource :: IPAddrSource
+    , apacheUserGetter :: Request -> Maybe BS.ByteString
+    , apacheRequestFilter :: Request -> Response -> Bool
+    }
+
+defaultApacheSettings :: ApacheSettings
+defaultApacheSettings = ApacheSettings
+    { apacheIPAddrSource = FromSocket
+    , apacheRequestFilter = \_ -> \_ -> True
```

Comment:
Would normally be `\_ _ -> True` IMO, but not really an issue.

Reply:
I'll fix that.